compiler/z/codegen/S390GenerateInstructions.cpp

```diff
@@ -3380,3 +3240,67 @@ TR::Instruction *generateZeroVector(TR::Node *node, TR::CodeGenerator *cg, TR::R
    {
    return generateVRIaInstruction(cg, TR::InstOpCode::VREPI, node, vecZeroReg, 0, 3);
    }
+
+#ifdef J9_PROJECT_SPECIFIC
+
+/**
+ * \brief This function returns true if the spcified instruction is designed to throw decimal overflow
+ * exceptions according to the POP.
+ *
+ * zLinux runs a process with this exception disabled. As a result of this, the JIT signal handler
+ * doesn't not receive any signal in case decimal overflow happens. This helper function can help
+ * codegen to decide whether to rely on condition code in such cases.
+*/
+bool
+isCanThrowDecimalOverflowException(TR::InstOpCode::Mnemonic op)
+   {
+   switch(op)
+      {
+   case TR::InstOpCode::AP:
+   case TR::InstOpCode::SP:
+   case TR::InstOpCode::SRP:
+   case TR::InstOpCode::ZAP:
+       return true;
+   default:
+       return false;
+      }
+   }
+
+void
+generateS390DAAExceptionRestoreSnippet(TR::CodeGenerator* cg,
+                                       TR::Node* n,
+                                       TR::Instruction* instr,
+                                       TR::InstOpCode::Mnemonic op,
+                                       bool isHasNOP)
```

Comment:
Similarly here, `is` and `Has` is confusing. Perhaps just `hasNOP`, or even better `generateNOP`. In either case this function needs to be documented as it is not immediately clear what it does.